includes/class-llms-sessions.php

```diff
@@ -0,0 +1,416 @@
+<?php
+/**
+ * User event session management.
+ *
+ * @package  LifterLMS/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * LLMS_Sessions class..
+ *
+ * @since [version]
+ */
+class LLMS_Sessions {
+
+	/**
+	 * Singleton instance
+	 *
+	 * @var  null
+	 */
+	protected static $_instance = null;
+
+	/**
+	 * Current user id.
+	 *
+	 * @var null
+	 */
+	protected $user_id = null;
+
+	/**
+	 * Get Main Singleton Instance.
+	 *
+	 * @since [version]
+	 *
+	 * @return LLMS_Sessions
+	 */
+	public static function instance() {
+		if ( is_null( self::$_instance ) ) {
+			self::$_instance = new self();
+		}
+		return self::$_instance;
+	}
+
+	/**
+	 * Private Constructor.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	private function __construct() {
+
+		add_filter( 'cron_schedules', array( $this, 'add_cron_schedule' ) );
+
+		if ( ! wp_next_scheduled( 'llms_end_idle_sessions' ) ) {
+			wp_schedule_event( time(), 'every_five_mins', 'llms_end_idle_sessions' );
+		}
+		add_action( 'llms_end_idle_sessions', array( $this, 'end_idle_sessions' ) );
+
+	}
+
+	/**
+	 * Add cron schedule for session end interval checks.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $schedules Array of cron schedules
+	 * @return  array
+	 */
+	public function add_cron_schedule( $schedules ) {
+
+		// Adds every 5 minutes to the existing schedules.
+		$schedules['every_five_mins'] = array(
```

Comment:
maybe prefix this schedule key in the future to possibly avoid 3rd parties code overriding this?!